docs/reference/mapping/types/flattened.asciidoc

```diff
@@ -191,6 +191,164 @@ POST my-index-000001/_search
 // TESTRESPONSE[s/"max_score" : 1.0/"max_score" : $body.hits.max_score/]
 // TESTRESPONSE[s/"_score" : 1.0/"_score" : $body.hits.hits.0._score/]
 
+You can also retrieve values from sub-fields of flattened fields with a
+<<runtime-retrieving-fields,runtime field>>. Instead of including
+`doc['<field_name>'].value` in your Painless script, use
+`doc['<field_name>.<sub-field_name>'].value`. For example, if you have a
+flattened field called `label` with a `release` sub-field, the Painless script
+in your runtime field would be `doc['labels.release'].value`.
```

Comment:
I am still on the fence, sorry! My main question is: "what is specific to runtime fields in how you access flattened sub-fields?". If we document how they are accessed, that should apply to every search section and not require specific documentation for each of them? Otherwise we need to mention how to access them from a query, from an aggregation, how they can be retrieved, how they can be accessed from a search script, and how from a runtime field. I see how maybe the doc['field.subfield'] is not super intuitive, is that what you are worried about? Should we then specify "this is how you access a flattened subfield from a script"?

Reply:
Yeah, I think it's more that we describe how to access sub-field values from a flattened field using a script. I used runtime fields as an example, though it could be any Painless script. 

I don't want to create a slippery slope where we're adding every possible way to access sub-fields, but I think that indicating how to retrieve these values with a script can be useful. What do you think about reframing this section around Painless scripts, but using a runtime field as an example (but de-emphasize runtime fields)? I agree with your assertion, and think we could say something like:

>"You can also use a Painless script to retrieve values from sub-fields of flattened fields. This capability means you can include your script in a runtime field to retrieve sub-field values."

Reply:
I am good with that plan but maybe we could only show a script example instead of the whole example?

Reply:
++ I just pushed changes in a6703a86cc913230f382ad306015e4eea8445e7c to remove the full example and focus on the script.